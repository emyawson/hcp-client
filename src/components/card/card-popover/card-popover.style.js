import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { breakpoints, colors, spacing, fontSize } from 'src/core';
import { weight } from 'src/components/fonts/weights';
import { combineRems, invertRem } from 'src/utils';

export const PopoverContainerDiv = styled.div`
  position: absolute;
  top: ${combineRems(spacing.four, spacing.two)};
  right: ${combineRems(spacing.five, spacing.two)};

  @media (min-width: ${breakpoints.medium}) {
    top: ${combineRems(spacing.four, spacing.three)};
    right: ${combineRems(spacing.five, spacing.two, spacing.one)};
  }

  @media (min-width: ${breakpoints.large}) {
    top: ${combineRems(spacing.five, invertRem(spacing.two))};
    right: ${combineRems(spacing.five, spacing.three, spacing.two)};
  }
`;

export const PopoverHeaderDiv = styled.div`
  color: ${colors.charcoal};
  font-size: ${fontSize.title};
  font-weight: ${weight.semiBold};
  border-bottom: 1px solid ${colors.silverMedium};
  padding-bottom: ${spacing.three};
  margin: ${spacing.four} ${spacing.four} 0;
`;

export const PopoverListUl = styled.ul`
  list-style: none;
  font-weight: ${weight.semiBold};
  font-size: ${fontSize.p};
  margin: 0;
  padding: 0 0 ${spacing.three};
`;

export const PopoverListItemLi = styled.li`
  padding: 0 ${spacing.four};
  border-left: 3px solid ${colors.white};

  &:hover {
    background-color: ${colors.silverLight};
    border-left: 3px solid ${colors.silverLight};

    &:active {
      border-left: 3px solid ${colors.brandBlue};
    }
  }
`;

export const PopoverItemLinkA = styled(Link)`
  display: inline-block;
  width: 100%;
  color: ${colors.grayDark};
  text-decoration: none;
  padding: 1rem 0;

  &:hover {
    color: ${colors.charcoal};
  }

  &:active {
    color: ${colors.brandBlue};
  }
`;

export const PopoverItemLinkTextSpan = styled.span`
  text-decoration: none;
`;
