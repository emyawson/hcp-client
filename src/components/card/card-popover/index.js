export * from './card-popover.component';
