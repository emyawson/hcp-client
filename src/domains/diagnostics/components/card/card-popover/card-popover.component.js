import React from 'react';

import { LocalizedText, Popover } from 'src/domains/diagnostics/components';

import {
  PopoverContainerDiv,
  PopoverHeaderDiv,
  PopoverListUl,
  PopoverListItemLi,
  PopoverItemLinkA,
  PopoverItemLinkTextSpan,
} from './card-popover.style';

type Props = {
  show: boolean,
  popoverTitleKey?: string,
  popoverActions?: {
    textKey: string,
    url: string,
  },
  innerRef?: () => void,
};

const getPopoverActionsList = actions => (
  <PopoverListUl>
    {actions.map(({ textKey, url }) => (
      <PopoverListItemLi key={`popover-action-${textKey}`}>
        <PopoverItemLinkA to={url}>
          <PopoverItemLinkTextSpan>
            <LocalizedText textKey={textKey} />
          </PopoverItemLinkTextSpan>
        </PopoverItemLinkA>
      </PopoverListItemLi>
    ))}
  </PopoverListUl>
);

export const CardPopover = ({
  show,
  popoverTitleKey,
  popoverActions,
  innerRef,
}: Props) => (
  <PopoverContainerDiv innerRef={innerRef}>
    <Popover pushLeft={0} show={show} width={17}>
      <PopoverHeaderDiv>
        <LocalizedText textKey={'userDropdown.managePatientProfile'} />
      </PopoverHeaderDiv>
      {getPopoverActionsList(popoverActions)}
    </Popover>
  </PopoverContainerDiv>
);

CardPopover.defaultProps = {
  popoverTitleKey: '',
  popoverActions: [],
  innerRef: () => null,
};
