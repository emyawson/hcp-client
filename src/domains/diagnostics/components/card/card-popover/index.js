export { CardPopover } from './card-popover.component';
