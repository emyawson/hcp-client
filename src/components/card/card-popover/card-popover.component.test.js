import React from 'react';
import { shallow } from 'enzyme';

import { CardPopover } from './card-popover.component';

describe('CardPopover', () => {
  const defaultProps = {
    show: true,
    popoverTitleKey: 'dashboard.patientCardPopover.header',
    popoverActions: [
      { textKey: 'dashboard.patientCardPopover.timePeriods', url: '/' },
    ],
  };

  it('renders correctly', () => {
    const wrapper = shallow(<CardPopover {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
