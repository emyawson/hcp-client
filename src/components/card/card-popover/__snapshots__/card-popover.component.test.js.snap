// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPopover renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    innerRef={[Function]}
    popoverActions={
      Array [
        Object {
          "textKey": "dashboard.patientCardPopover.timePeriods",
          "url": "/",
        },
      ]
    }
    popoverTitleKey="dashboard.patientCardPopover.header"
    show={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown
        pushLeft={0}
        show={true}
        width={17}
      >
        <styled.div>
          <LocalizedText
            textKey="userDropdown.managePatientProfile"
          />
        </styled.div>
        <styled.ul>
          <styled.li>
            <Styled(Link)
              to="/"
            >
              <styled.span>
                <LocalizedText
                  textKey="dashboard.patientCardPopover.timePeriods"
                />
              </styled.span>
            </Styled(Link)>
          </styled.li>
        </styled.ul>
      </Unknown>,
      "innerRef": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <styled.div>
            <LocalizedText
              textKey="userDropdown.managePatientProfile"
            />
          </styled.div>,
          <styled.ul>
            <styled.li>
              <Styled(Link)
                to="/"
              >
                <styled.span>
                  <LocalizedText
                    textKey="dashboard.patientCardPopover.timePeriods"
                  />
                </styled.span>
              </Styled(Link)>
            </styled.li>
          </styled.ul>,
        ],
        "pushLeft": 0,
        "show": true,
        "width": 17,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <LocalizedText
              textKey="userDropdown.managePatientProfile"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "textKey": "userDropdown.managePatientProfile",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.li>
                <Styled(Link)
                  to="/"
                >
                  <styled.span>
                    <LocalizedText
                      textKey="dashboard.patientCardPopover.timePeriods"
                    />
                  </styled.span>
                </Styled(Link)>
              </styled.li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "popover-action-dashboard.patientCardPopover.timePeriods",
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Link)
                  to="/"
                >
                  <styled.span>
                    <LocalizedText
                      textKey="dashboard.patientCardPopover.timePeriods"
                    />
                  </styled.span>
                </Styled(Link)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <styled.span>
                    <LocalizedText
                      textKey="dashboard.patientCardPopover.timePeriods"
                    />
                  </styled.span>,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <LocalizedText
                      textKey="dashboard.patientCardPopover.timePeriods"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "textKey": "dashboard.patientCardPopover.timePeriods",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown
          pushLeft={0}
          show={true}
          width={17}
        >
          <styled.div>
            <LocalizedText
              textKey="userDropdown.managePatientProfile"
            />
          </styled.div>
          <styled.ul>
            <styled.li>
              <Styled(Link)
                to="/"
              >
                <styled.span>
                  <LocalizedText
                    textKey="dashboard.patientCardPopover.timePeriods"
                  />
                </styled.span>
              </Styled(Link)>
            </styled.li>
          </styled.ul>
        </Unknown>,
        "innerRef": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <styled.div>
              <LocalizedText
                textKey="userDropdown.managePatientProfile"
              />
            </styled.div>,
            <styled.ul>
              <styled.li>
                <Styled(Link)
                  to="/"
                >
                  <styled.span>
                    <LocalizedText
                      textKey="dashboard.patientCardPopover.timePeriods"
                    />
                  </styled.span>
                </Styled(Link)>
              </styled.li>
            </styled.ul>,
          ],
          "pushLeft": 0,
          "show": true,
          "width": 17,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <LocalizedText
                textKey="userDropdown.managePatientProfile"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "textKey": "userDropdown.managePatientProfile",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.li>
                  <Styled(Link)
                    to="/"
                  >
                    <styled.span>
                      <LocalizedText
                        textKey="dashboard.patientCardPopover.timePeriods"
                      />
                    </styled.span>
                  </Styled(Link)>
                </styled.li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "popover-action-dashboard.patientCardPopover.timePeriods",
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Link)
                    to="/"
                  >
                    <styled.span>
                      <LocalizedText
                        textKey="dashboard.patientCardPopover.timePeriods"
                      />
                    </styled.span>
                  </Styled(Link)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <styled.span>
                      <LocalizedText
                        textKey="dashboard.patientCardPopover.timePeriods"
                      />
                    </styled.span>,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": <LocalizedText
                        textKey="dashboard.patientCardPopover.timePeriods"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "textKey": "dashboard.patientCardPopover.timePeriods",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
